### PR TITLE
Disable TE cross entropy loss fusion

### PR DESCRIPTION
**Pull request by mchrzanowski**
[X] I, the PR author, have personally reviewed every line of this PR.

  # What does this PR do ?

  Disables the Transformer Engine implementation of cross entropy loss fusion with an assertion due to observed training stability issues, while keeping native cross entropy fusion available.

  ## Issue tracking

  Linked issue: N/A, small stability bug fix.

  ## Contribution process

  ### Pre-checks

  - [X] I have added relevant unit tests
  - [ ] I have added relevant functional tests
  - [X] I have added proper typing to my code [Typing guidelines](https://docs.python.org/3/library/typing.html)
  - [X] I have added relevant documentation
  - [x] I have run the [autoformatter.sh](https://github.com/NVIDIA/Megatron-LM/blob/main/tools/autoformat.sh) on my PR

  ### Validation

  - Added a unit test covering rejection of `cross_entropy_loss_fusion=True` with `cross_entropy_fusion_impl='te'`.
  - Updated existing functional test configs and MoE README guidance to use `native`.
  - Ran `python -m py_compile` on changed Python files.
  - Ran `git diff --check`.
  - Could not run pytest in the local environment because `torch` is not installed.

### CODE REVIEW

**Review by yashaswikarnati — APPROVED** [review]

**Review by kvareddy — APPROVED** [review]

**Review by yaox12 — APPROVED** [review]

**Review by yaoyu-33 — APPROVED** [review]

**Comment by ko3n1g** [comment]
/ok to test [eab65fb](https://github.com/NVIDIA/Megatron-LM/pull/5115/commits/eab65fbb31c32271d1ccf807e8fedc5036ff3c25)

**Review by jaredcasper — APPROVED** [review]
